src/snackbar/snackbar.js

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function MaterialSnackbar(element) {
+  'use strict';
+
+  this.element_ = element;
+  this.active = false;
+  this.init();
+
+}
+
+MaterialSnackbar.prototype.defaults = {
+  timeout: 8000
+};
+
+MaterialSnackbar.prototype.cssClasses = {
+  snackbar: 'mdl-snackbar',
+  message: 'mdl-snackbar__text',
+  action: 'mdl-snackbar__action',
+  activeSnackbar: 'is-active'
+};
+
+MaterialSnackbar.prototype.queuedNotifications = [];
+
+MaterialSnackbar.prototype.setMessage = function(text) {
+  'use strict';
+  this.message_ = text;
+};
+
+MaterialSnackbar.prototype.setTimeout = function(time) {
+  'use strict';
+  this.timeout_ = parseInt(time);
+};
+
+MaterialSnackbar.prototype.setActionText = function(text) {
+  'use strict';
+  this.actionText_ = text;
+};
+
+MaterialSnackbar.prototype.setActionHandler = function(handler) {
+  'use strict';
+  this.actionHandler_ = handler;
+};
+
+MaterialSnackbar.prototype.createSnackbar = function() {
+  'use strict';
+  this.snackbarElement_ = document.createElement('div');
+  this.textElement_ = document.createElement('div');
+  this.snackbarElement_.classList.add(this.cssClasses.snackbar);
+  this.textElement_.classList.add(this.cssClasses.message);
+  this.snackbarElement_.appendChild(this.textElement_);
+  this.snackbarElement_.setAttribute('aria-hidden', true);
+
+  if (this.actionHandler_) {
+    this.actionElement_ = document.createElement('button');
+    this.actionElement_.type = 'button';
+    this.actionElement_.classList.add(this.cssClasses.action);
+    this.actionElement_.textContent = this.actionText_;
+    this.snackbarElement_.appendChild(this.actionElement_);
+    this.actionElement_.addEventListener('click', this.actionHandler_);
+  }
+
+  this.element_.appendChild(this.snackbarElement_);
+
+  this.textElement_.textContent = this.message_;
+  this.snackbarElement_.classList.add(this.cssClasses.activeSnackbar);
+  this.snackbarElement_.setAttribute('aria-hidden', false);
+  setTimeout(this.cleanup.bind(this), this.timeout_);
+
+};
+
+MaterialSnackbar.prototype.removeSnackbar = function() {
+  'use strict';
+  if (this.actionElement_ && this.actionElement_.parentNode) {
+    this.actionElement_.parentNode.removeChild(this.actionElement_);
+  }
+  this.textElement_.parentNode.removeChild(this.textElement_);
+  this.snackbarElement_.parentNode.removeChild(this.snackbarElement_);
+};
+
+MaterialSnackbar.prototype.showSnackbar = function(data) {
```

Comment:
Data is required, otherwise we can't show anything. I guess we should do a check and just `return` as early as possible if nothing was provided, or throw an error (which would be my preference to reduce debugging load.)